diff --git a/nemo_gym/global_config.py b/nemo_gym/global_config.py
--- a/nemo_gym/global_config.py
+++ b/nemo_gym/global_config.py
@@ -404,6 +404,24 @@ class GlobalConfigDictParser(BaseModel):
 
         self._recursively_swap_keys(global_config_dict)
 
+        # TODO @bxyu-nvidia: We need a better way of handling dummy model configs
+        with open_dict(global_config_dict):
+            for top_level_value in global_config_dict.values():
+                if not (
+                    isinstance(top_level_value, (DictConfig))
+                    and "responses_api_models" in top_level_value
+                    # We check `len(top_level_value) > 1` in case the policy model is inherited from.
+                    and (len(top_level_value) > 1 or len(top_level_value["responses_api_models"]) > 1)
+                    and "dummy_model" in top_level_value["responses_api_models"]
+                ):
+                    continue
+
+                dummy_value = top_level_value["responses_api_models"].pop("dummy_model")
+                actual_key = next(iter(top_level_value["responses_api_models"]))
+                top_level_value["responses_api_models"][actual_key] = OmegaConf.merge(
+                    dummy_value, top_level_value["responses_api_models"][actual_key]
+                )
+
         # Almost-server detection and reporting
         almost_servers = self.detect_and_report_almost_servers(global_config_dict)
 
